semanticdb/semanticdb3/javacp.md

```diff
@@ -0,0 +1,480 @@
+        <li><code>int[]</code> ~ <code>TypeRef(None, &lt;scala.Array#&gt;, List(&lt;scala.Int#&gt;))</code>.</li>
+      </ul>
+    </td>
+  </tr>
+  <tr>
+    <td valign="top">Intersection types [<a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-4.html#jls-4.9">81</a>]</td>
+    <td>
+      <ul>
+        <li><code>T1 &amp; ... &amp; Tn</code> ~ <code>IntersectionType(&lt;T1&gt;, ..., &lt;Tn&gt;)</code>.</li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+Notes:
+* Primitive and array types are converted to their equivalent [Scala type](#scala-type) representations.
+* Wildcard type arguments of parameterized types are currently incorrectly represented as `local_wildcard`,
+  this behavior is expected to change in the future and should not be relied upon.
+* Method throws clauses are discarded.
+
+<a name="java-symbolinformation"></a>
+#### SymbolInformation
+
+```protobuf
+message SymbolInformation {
+  reserved 2, 6, 7, 8, 9, 10, 12;
+  string symbol = 1;
+  Language language = 16;
+  Kind kind = 3;
+  int32 properties = 4;
+  string name = 5;
+  Type tpe = 11;
+  repeated Annotation annotations = 13;
+  Accessibility accessibility = 14;
+  string owner = 15;
+}
+```
+
+<table>
+  <tr>
+    <td><b>Field</b></td> <td><b>Explanation</b></td>
+  </tr>
+  <tr>
+    <td><code>symbol</code></td>
+    <td>See <a href="#java-symbol">Symbol</a>.</td>
+  </tr>
+  <tr>
+    <td><code>language</code></td>
+    <td><code>JAVA</code>.</td>
+  </tr>
+  <tr>
+    <td><code>kind</code></td>
+    <td>Explained below on per-definition basis.</td>
+  </tr>
+  <tr>
+    <td><code>properties</code></td>
+    <td>Explained below on per-definition basis.</td>
+  </tr>
+  <tr>
+    <td><code>name</code></td>
+    <td>See <a href="#java-symbol">Symbol</a>.</td>
+  </tr>
+  <tr>
+    <td><code>tpe</code></td>
+    <td>Explained below on per-definition basis.</td>
+  </tr>
+  <tr>
+    <td><code>annotations</code></td>
+    <td>Explained below on per-definition basis.</td>
+  </tr>
+  <tr>
+    <td><code>accessibility</code></td>
+    <td>Explained below on per-definition basis.</td>
+  </tr>
+  <tr>
+    <td><code>owner</code></td>
+    <td>See <a href="#java-symbol">Symbol</a>.</td>
+  </tr>
+</table>
+
+**Class declarations** [\[76\]][76] are represented by a single symbol with the `CLASS` kind.
+
+```java
+package a;
+class C extends S1 implements I {
+  String m1;
+  static unit m2();
+  int m3(String e) throws E;
+  static class D1<T1 extends S2 & S3, T2> { }
+  class D2 { }
+}
+```
+
+<table>
+  <tr>
+    <td><b>Definition</b></td>
+    <td width="275px"><b>Symbol</b></td>
+    <td><b>Kind</b></td>
+    <td><b>Signature</b></td>
+  </tr>
+  <tr>
+    <td><code>C</code></td>
+    <td><code>a.C#</code></td>
+    <td><code>CLASS</code></td>
+    <td><code>ClassInfoType(List(), List(&lt;S1&gt;, &lt;I&gt;), List(&lt;a.C#m1&gt;, &lt;a.C#m3(String).&gt;, &lt;a.C#D2#&gt;, &lt;a.C#m2().&gt;, &lt;a.C#D1#&gt;))</code></td>
```

Comment:
It's enough to say `List(<m1>, <m3>, ...)` instead of writing symbols for those members.